sdk/core/azure-core/test/ut/url_test.cpp

```diff
@@ -333,4 +333,49 @@ namespace Azure { namespace Core { namespace Test {
     EXPECT_NE(params.find("param"), params.end());
     EXPECT_EQ(params["param"], "value");
   }
+
+  TEST(URL, LeadingSlashInPath)
+  {
+    Core::Url const u0("https://www.microsoft.com");
+    Core::Url const u1("https://www.microsoft.com/");
+
+    EXPECT_EQ(u0.GetAbsoluteUrl(), "https://www.microsoft.com");
+    EXPECT_EQ(u1.GetAbsoluteUrl(), "https://www.microsoft.com");
```

Comment:
In this case, us dropping the trailing `/` is a bit of a strange behavior.

If we are to normalize the two, we should do it the other way and have the trailing `/` in both.

For example, here's the behavior in .NET:
```C#
Uri ur0 = new Uri("http://www.microsoft.com");
Uri ur1 = new Uri("http://www.microsoft.com/");
Console.WriteLine(ur0); // http://www.microsoft.com/
Console.WriteLine(ur1); // http://www.microsoft.com/

Console.WriteLine(ur0.AbsolutePath); // /
Console.WriteLine(ur1.AbsolutePath); // /

Console.WriteLine(ur0.AbsoluteUri); // http://www.microsoft.com/
Console.WriteLine(ur1.AbsoluteUri); // http://www.microsoft.com/
```

.NET also has different semantics for when a path is "appended" versus when it is "replaced" based on whether a `/` is present or not.
https://learn.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-8.0#system-uri-ctor(system-uri-system-string)
```C#
Uri ur0 = new Uri("http://www.microsoft.com/originalPath");
Uri ur1 = new Uri("http://www.microsoft.com/originalPath/");

Uri ur2 = new Uri(ur0, "path");
Uri ur3 = new Uri(ur0, "/path");

Uri ur4 = new Uri(ur1, "path");
Uri ur5 = new Uri(ur1, "/path");

Console.WriteLine(ur2); // http://www.microsoft.com/path
Console.WriteLine(ur3); // http://www.microsoft.com/path
Console.WriteLine(ur4); // http://www.microsoft.com/originalPath/path
Console.WriteLine(ur5); // http://www.microsoft.com/path
```

Reply:
We could update it, but I am not sure about the implications, if it is potentially breaking change. Feel free to open a work item for that.
For potentially breaking stuff, I would add a set of different methods to be .NET compatible, or even implement Uri (not Url) class which implements semantics that is 100% compatible with .NET.
Having different behavior, when it is a constructor parameter, is less of a concern, because it is a constructor parameter -i.e. clearly different semantics. If we want to, se can add a constructor overload which would behave the same. But that is a different work item.